unitTests/Repartitioner.C

```diff
@@ -0,0 +1,630 @@
+}
+
+TEST_P(RepartitionerFixture1D, has_correct_properties_for_n_rank)
+{
+    // Arrange
+    label ranks_per_gpu = GetParam();
+    // Act
+    auto repartitioner = Repartitioner(local_size, ranks_per_gpu, 0, exec);
+
+    // Assert
+    EXPECT_EQ(repartitioner.is_owner(exec),
+              (rank % ranks_per_gpu == 0) ? true : false);
+
+    EXPECT_EQ(
+        repartitioner.compute_repart_size(local_size, ranks_per_gpu, exec),
+        (repartitioner.is_owner(exec) ? ranks_per_gpu * local_size : 0));
+}
+
+TEST_P(RepartitionerFixture2D, can_convert_to_global)
+{
+    // Arrange
+    using namespace gko::experimental::distributed;
+    auto ref_exec = exec.get_ref_exec();
+    auto partition = gko::share(
+        build_partition_from_local_size<label, label>(ref_exec, comm, 4));
+
+    vec_vec exp_global_idxs{
+        {4, 6, 8, 9},    // rank 0
+        {1, 3, 12, 13},  // rank 1
+        {2, 3, 12, 14},  // rank 2
+        {6, 7, 9, 11}    // rank 3
+    };
+
+    // Act
+    auto global_idxs = detail::convert_to_global(partition, idxs[rank],
+                                                 spans[rank], ranks[rank]);
+
+    // Assert
+    EXPECT_EQ(global_idxs, exp_global_idxs[rank]);
+}
+
+
+TEST_P(RepartitionerFixture1D, can_exchange_spans_and_ranks_for_n_ranks)
+{
+    // Arrange
+    label ranks_per_gpu = GetParam();
+
+    std::vector<gko::span> spans{{0, 5}, {5, 10}};
+    std::vector<label> ranks{rank + 1, rank + 2};
+
+    std::map<label, vec_vec> exp_ranks;
+    exp_ranks.emplace(1, vec_vec{{1, 2}, {2, 3}, {3, 4}, {4, 5}});
+    exp_ranks.emplace(2, vec_vec{{1, 2, 2, 3}, {}, {3, 4, 4, 5}, {}});
+    exp_ranks.emplace(4, vec_vec{{1, 2, 2, 3, 3, 4, 4, 5}, {}, {}, {}});
+
+    std::map<label, vec_vec> exp_spans_begin;
+    exp_spans_begin.emplace(1, vec_vec{{0, 5}, {0, 5}, {0, 5}, {0, 5}});
+    exp_spans_begin.emplace(2, vec_vec{{0, 5, 10, 15}, {}, {0, 5, 10, 15}, {}});
+    exp_spans_begin.emplace(
+        4, vec_vec{{0, 5, 10, 15, 20, 25, 30, 35}, {}, {}, {}});
+    std::map<label, vec_vec> exp_spans_end;
+    exp_spans_end.emplace(1, vec_vec{{5, 10}, {5, 10}, {5, 10}, {5, 10}});
+    exp_spans_end.emplace(2, vec_vec{{5, 10, 15, 20}, {}, {5, 10, 15, 20}, {}});
+    exp_spans_end.emplace(4,
+                          vec_vec{{5, 10, 15, 20, 25, 30, 35, 40}, {}, {}, {}});
+
+    // Act
+    auto [new_spans, new_ranks, origins] =
+        detail::exchange_span_ranks(exec, ranks_per_gpu, spans, ranks);
+
+    std::vector<label> res_spans_begin{};
+    std::vector<label> res_spans_end{};
+
+    for (auto span : new_spans) {
+        res_spans_begin.push_back(span.begin);
+        res_spans_end.push_back(span.end);
+    }
+
+    // Assert
+    ASSERT_EQ(new_ranks, exp_ranks[ranks_per_gpu][rank]);
+    ASSERT_EQ(res_spans_begin, exp_spans_begin[ranks_per_gpu][rank]);
+    ASSERT_EQ(res_spans_end, exp_spans_end[ranks_per_gpu][rank]);
+}
+
+TEST_P(RepartitionerFixture1D, can_repartition_sparsity_pattern_1D_for_n_ranks_unfused)
```

Comment:
What is the difference between the unit test for unfused case and the one for fused case?

Reply:
fused means that all interfaces/submatrices are recombined into a single local and non-local matrix